llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -304,7 +304,11 @@ void SPIRVModuleAnalysis::collectFuncNames(MachineInstr &MI,
     Register GlobalReg = MAI.getRegisterAlias(MI.getMF(), Reg);
     assert(GlobalReg.isValid());
     // TODO: check that it does not conflict with existing entries.
-    MAI.FuncNameMap[F.getGlobalIdentifier()] = GlobalReg;
+    GlobalValue::LinkageTypes Linkage = F.getLinkage();
+    StringRef Name = F.hasName() ? F.getName() : ".anonymous";
+    StringRef ModuleFileName = F.getParent()->getSourceFileName();
```

Comment:
This code is duplicated in llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.h:158, probably we need to define a function like this `std::string getFunctionGlobalIdentifier(const Function *F)`.